apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+        var employee = employees[employeeIds[msg.sender]];
+        require(getTimestamp() > employee.lastAllocation && getTimestamp() - employee.lastAllocation > 15768000); // half a year in seconds
+
+        // check arrays match
+        require(tokens.length == distribution.length);
+
+        // check distribution is right
+        uint256 sum = 0;
+        uint256 i;
+        for (i = 0; i < distribution.length; i++) {
+            // check token is allowed
+            require(employee.allowedTokens[tokens[i]]);
+            // set distribution
+            employee.allocation[tokens[i]] = distribution[i];
+            sum = sum.add(distribution[i]);
+        }
+        require(sum == 100);
+
+        employee.lastAllocation = getTimestamp();
+    }
+
+    /**
+     * @dev Get payment proportion for a token and an employee (the caller)
+     * @notice Get payment proportion for a token and an employee (the caller)
+     * @param token The token address
+     */
+    function getAllocation(address token) public constant auth(EMPLOYEE_ROLE) returns(uint256 allocation) {
+        var employee = employees[employeeIds[msg.sender]];
+        allocation = employee.allocation[token];
+    }
+
+    /**
+     * @dev payday To withdraw monthly payment by employee (the caller)
+     * @notice Only callable once a month. Assumed token has these standard checks implemented:
+     *         https://theethereum.wiki/w/index.php/ERC20_Token_Standard#How_Does_A_Token_Contract_Work.3F
+     */
+    function payday() public auth(EMPLOYEE_ROLE) {
+        var employee = employees[employeeIds[msg.sender]];
+        // check that employee exists (and matches)
+        require(employee.accountAddress == msg.sender);
+        require(getTimestamp() > employee.lastPayroll && getTimestamp() - employee.lastPayroll > 2628000); // 1/12 year in seconds
+
+        // loop over allowed tokens
+        bool somethingPaid = false;
+        for (uint i = 0; i < employee.allowedTokensArray.length; i++) {
+            var token = employee.allowedTokensArray[i];
+            if (employee.allocation[token] == 0)
+                continue;
+            require(checkExchangeRate(token));
+            uint256 tokenAmount = employee.yearlyUSDSalary
+                .mul(employee.allocation[token]).mul(exchangeRates[token]) / 1200;
+            // we could make a recurring payment, but it would change the flow and we should take care of changes in allocation
+            ERC20 tokenContract = ERC20(token);
+            finance.newPayment(
+                tokenContract,
+                msg.sender,
+                tokenAmount,
+                0,
+                0,
+                1,
+                ""
+            );
+            LogSendPayroll(msg.sender, token, tokenAmount);
+            somethingPaid = true;
+        }
+        require(somethingPaid);
+        // finally update last payroll date
+        employee.lastPayroll = getTimestamp();
```

Comment:
We should set it to `employee.lastPayroll += 1 year / 12`, because otherwise taking some time to send the `payday()` transaction will be making the employee lose money. This will be specially important when we give this contract the ability to have variable pay intervals :)

Reply:
Absolutely